docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md

```diff
@@ -0,0 +1,204 @@
+
+![Create an Empty Job](../../../images/azure/create-empty-job.png)
+
+On the next screen, select an [agent pool](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/pools-queues?view=vsts) to run the job.
+
+> Important! The agent should run on a machine with all the necessary browsers installed.
+
+![Select Agent Pool](../../../images/azure/select-agent-pool.png)
+
+Now you can add tasks to the job. Click the **+** button to create a new task.
+
+![Create New Task](../../../images/azure/add-new-task.png)
+
+Select **Node Tool Installer** from the task list. This task installs Node.js and adds it to the `PATH` variable.
+
+![Add Node.js Installer](../../../images/azure/add-node-installer.png)
+
+The task is now added to the pipeline. Click its name in the pipeline to enter the task settings.
+
+On the settings screen, specify the Node.js version to install. Use the latest LTS version - you can find its number on the [Node.js website](https://nodejs.org/en/).
+
+![Specify Node.js Version](../../../images/azure/set-node-version.png)
+
+Now add another task that runs a shell command. Click the **+** button again and select **Command Line** from the task list.
+
+![Add a Command Line Task](../../../images/azure/add-command-line.png)
+
+Configure this command to install TestCafe. Open its settings, specify a meaningful name and enter the following command:
+
+```cmd
+npm install
+```
+
+![Install TestCafe](../../../images/azure/npm-install.png)
+
+In the same manner, add another **Command Line** task that runs TestCafe tests.
+
+```cmd
+npm test
+```
+
+![Run TestCafe Tests](../../../images/azure/npm-test.png)
+
+Save the changes and run this task using the **Save & queue** button.
+
+![Save Pipeline](../../../images/azure/save-pipeline.png)
+
+This opens the **Save build pipeline and queue** window. Specify an optional comment that describes the changes you have made. Select an agent pool that runs this task, the repository branch and the commit that contains code to build. You can leave the **Commit** field blank to use the latest commit. Click **Save & queue**.
+
+![Save Pipeline](../../../images/azure/save-options.png)
+
+## Step 4 - Run Test Task and View Results
+
+### Option 1 - Enable Continuous Integration
+
+You can set Azure DevOps to run the pipeline every time your code changes. To do this, you need to enable continuous integration.
+
+On Azure Pipelines, continuous integration is enabled for every newly created build.
+
+On Azure DevOps Server, select the build and click **Edit**.
+
+![Edit Build](../../../images/azure/edit-task-server.png)
+
+Then open the **Triggers** tab.
+
+![Triggers Tab](../../../images/azure/set-ci.png)
+
+Click the repository name under the **Continuous integration** header and check **Enable continuous integration**.
+
+![Triggers Tab](../../../images/azure/enable-ci.png)
+
+If you need to build a specific branch or path, you can add **Branch filters** and/or **Path filters**.
```

Comment:
If you need to build a specific branch or path, you can add **Branch filters** and/or **Path filters** -> To build a specific branch or path, add **Branch filters** and/or **Path filters**.